sbc_email/models/import_letters_history.py

```diff
@@ -379,28 +380,109 @@ def _get_smb_connection(self):
             return SMBConnection(
                 SmbConfig.smb_user, SmbConfig.smb_pass, 'openerp', 'nas')
 
-    @api.model
-    def import_web_letter(self):
+    @api.one
+    def import_web_letter(self, child_code, sponsor_ref, language_name,
+                          original_text, template_name, pdf_letter=None):
         """
         Call when a letter is set on web site:
             - add web letter to an import set with import letter config
               'Web letter'
         """
-        import_web_letter = self.search([
-            ('config_id.name', '=', 'Web Letter'),
-            ('state', '!=', 'done')])
+        child_id = None
+        if child_code:
+            # Retrieve child code and find corresponding id
+            model_child = self.env['compassion.child'].search(
+                [('code', '=', child_code)], limit=1)
+
+            child_id = model_child.id
+            logger.info("child id found {}".format(child_id))
+        else:
+            return False
+
+        sponsor_id = None
+        if sponsor_ref:
+            # Retrieve sponsor reference and find corresponding id
+            model_sponsor = self.env['res.partner'].search(
+                [('ref', '=', sponsor_ref)], limit=1)
+
+            sponsor_id = model_sponsor.id
+            logger.info("sponsor id found {}".format(sponsor_id))
+        else:
+            return False
+
+        lang_id = None
+        if language_name:
+            # Retrieve language name and find corresponding id
+            model_lang = self.env['res.lang.compassion'].search(
+                [('name', '=', language_name)], limit=1)
+
+            lang_id = model_lang.id
+            logger.info("language id found {}".format(lang_id))
+
+        if template_name:
+            # Retrieve template name and find corresponding id
+            model_template = self.env['correspondence.template'].search(
+                [('name', '=', template_name)], limit=1)
+
+            template = None
+
+            # TODO TO IMPROVE
+            template_id = model_template.id
+            logger.info("template id found {}".format(template_id))
+
+            template = model_template.browse(template_id)
+
+            if template:
+                logger.info("template found: {}".format(template.name))
+            else:
+                return False
```

Comment:
If the template is not found, instead of aborting the import you could use the default template `self.env.ref('sbc_compassion.default_template')` (if I remember correctly)